include/exiv2/matroskavideo.hpp

```diff
@@ -0,0 +1,143 @@
+// class definitions
+
+    // Todo: Should be hidden
+    /*!
+      @brief Helper structure for the Matroska tags lookup table.
+     */
+    struct MatroskaTags {
+        uint64_t val_;                          //!< Tag value
+        const char* label_;                     //!< Translation of the tag value
+
+        //! Comparison operator for use with the find template
+       bool operator==(uint64_t key) const { return val_ == key; }
+    }; // struct TagDetails
+
+    /*!
+      @brief Class to access Matroska video files.
+     */
+    class MatroskaVideo : public Image {
+    public:
+        //! @name Creators
+        //@{
+        /*!
+          @brief Constructor for a Matroska video. Since the constructor
+              can not return a result, callers should check the good() method
+              after object construction to determine success or failure.
+          @param io An auto-pointer that owns a BasicIo instance used for
+              reading and writing image metadata. \b Important: The constructor
+              takes ownership of the passed in BasicIo instance through the
+              auto-pointer. Callers should not continue to use the BasicIo
+              instance after it is passed to this method. Use the Image::io()
+              method to get a temporary reference.
+         */
+        MatroskaVideo(BasicIo::UniquePtr io);
+        //@}
+
+        //! @name Manipulators
+        //@{
+        void readMetadata() override;
+        void writeMetadata();
+        //@}
+
+        //! @name Accessors
+        //@{
+        std::string mimeType() const;
+        //@}
+
+    protected:
+        /*!
+          @brief Function used to calulate the size of a block.
+              This information is only stored in one byte.
+              The size of the block is calculated by counting
+              the number of leading zeros in the binary code of the byte.
+              Size = (No. of leading zeros + 1) bytes
+          @param b The byte, which stores the information to calculate the size
+          @return Return the size of the block.
+         */
+        uint32_t findBlockSize(byte b);
+        /*!
+          @brief Check for a valid tag and decode the block at the current IO position.
+              Calls contentManagement() or skips to next tag, if required.
+         */
+        void decodeBlock();
+        /*!
+          @brief Interpret tag information, and save it in the respective XMP container.
+          @param mt Pointer to current tag,
+          @param buf Pointer to the memory area with the tag information.
+          @param size Size of \em buf.
+         */
+        void contentManagement(const MatroskaTags* mt, const byte* buf, long size);
+        /*!
+          @brief Calculates Aspect Ratio of a video, and stores it in the
+              respective XMP container.
+         */
+        void aspectRatio();
+
+    private:
+        //! @name NOT Implemented
```

Comment:
Not implemented functions should be public and marked = delete;

Reply:
done here https://github.com/Exiv2/exiv2/pull/2370/commits/65a01dc171f4dc230eed22754990d4c25dd0078d

Reply:
hmm? I don't see = delete;

Reply:
my bad, It's here : 49f6acd